lib/concentrate/stop_time_update.ex

```diff
@@ -0,0 +1,80 @@
+defmodule Concentrate.StopTimeUpdate do
+  @moduledoc """
+  Structure for representing an update to a StopTime (e.g. a predicted arrival or departure)
+  """
+  import Concentrate.StructHelpers
+  alias Gtfs.Stop
+
+  defstruct_accessors([
+    :trip_id,
+    :stop_id,
+    :arrival_time,
+    :departure_time,
+    :stop_sequence,
+    :status,
+    :track,
+    :platform_id,
+    :uncertainty,
+    schedule_relationship: :SCHEDULED
+  ])
+
+  @doc """
+  Returns a time for the StopTimeUpdate: arrival if present, otherwise departure.
+  """
+  @spec time(%__MODULE__{}) :: non_neg_integer | nil
+  def time(%__MODULE__{arrival_time: time}) when is_integer(time), do: time
+  def time(%__MODULE__{departure_time: time}), do: time
+
+  @compile inline: [time: 1]
+
+  @doc """
+  Marks the update as skipped (when the stop is closed, for example).
+  """
+  @spec skip(%__MODULE__{}) :: t
+  def skip(%__MODULE__{} = stu) do
+    %{stu | schedule_relationship: :SKIPPED, arrival_time: nil, departure_time: nil}
+  end
+
+  defimpl Concentrate.Mergeable do
+    @default_key_opts [stop_fn: &Gtfs.stop/1]
+    def key(%{trip_id: trip_id, stop_id: stop_id, stop_sequence: stop_sequence}, opts) do
+      opts = Keyword.merge(@default_key_opts, opts)
+      stop_fn = opts[:stop_fn]
+
+      parent_station_id =
+        if stop_id do
+          stop_id
+          |> stop_fn.()
+          |> Stop.parent_station_id()
+        end
+
+      {trip_id, parent_station_id, stop_sequence}
```

Comment:
What's this key used for? Is `{trip_id, stop_sequence}` enough? If so, `Gtfs.stop` and `Stop.parent_station_id` aren't needed.

Reply:
That's true at the moment, but different features are going to want more pieces of data as we go. I don't know if any of that will need to come from StopTimeUpdate, but I'm relunctant to throw out data until we have a better sense.

Reply:
If we want predictions in the future, we could probably get them simpler just from the public GTFS-RT feed than by duplicating the functionality here. We wouldn't need the same merging from multiple sources logic that we do for VehiclePositions, so I don't think we should include them yet.

Reply:
I'm not sure I follow what you mean by the public GTFS-RT feed, is there something to that that is different than what we are currently feeding through this?

Either way, I think my comment about avoiding custom changes feels applicable here too.